packages/dd-trace/src/config.js

```diff
@@ -189,9 +189,9 @@ function validateNamingVersion (versionString) {
  * @param {string | string[]} input
  */
 function splitJSONPathRules (input) {
-  if (!input) return
+  if (!input || input.toLowerCase() === 'false' || input === '$') return
   if (Array.isArray(input)) return input
-  if (input === 'all') return []
+  if (input === 'all' || input.toLowerCase() === 'true') return []
   return input.split(',')
```

Comment:
Instead of validating the entries later during runtime, we can already validate the input right away and filter all invalid ones. That applies to both, the array and the string notation.

That way the runtime has less overhead and we would only log a warning once instead of on each usage.